# GFF Merge
I wrote this script to combine gff(3) files generated by BRAKER (https://github.com/Gaius-Augustus/BRAKER) and EarlGrey (https://github.com/TobyBaril/EarlGrey), to view potential TE accumulation within gene introns of intrest. Combines various functions of genometools (https://genometools.org/)

Additonally, there's an option to include a locus (that takes a certain format, see below), and draw a sketch of your locus annotation with the `AnnotationSketch` function of genometools. 

## Usage 
It's straightforward to run if you have nextflow installed. You provide 3 parameters, a species ID (for file name), a `braker_gff`(gff3), and a `repeat_gff`(gff3)
```shell
# merge your gffs
nextflow run artorias111/FeatureFlow --species_id hsapien134 --braker_gff /path/to/braker/braker.gff3 --repeat_gff /path/to/eg/repeats.gff

# merge your gffs, and also retrive a sketch of a certain region of interest
nextflow run artorias111/FeatureFlow --species_id hsapien134 --braker_gff /path/to/braker/braker.gff3 --repeat_gff /path/to/eg/repeats.gff --genome_region chr12-243-534
```

### Specifying the genome region
The genome region contains a chromosome, start co-ordinate (1-index), and stop co-ordinate, seperated by a single dash `-`. Make sure the chromsome number/ID matches the first column of the gff file(s). 

Examples: 
```shell
# chromosome 12 (written as chr12 in the gff file), from bp 243 to 534
chr12-243-534

# scaffold_9 from bp 434353 to 23124322
scaffold_9-434353-23124322
```
